### PR TITLE
Added: Copy text without dismissing the selection

### LINKED ISSUE #15371: Copy text without dismissing the selection
> In Notepad, if there is a selection, pressing Ctrl+C, the text will be copied to the clipboard, otherwise nothing happens. This is the same in other edit boxes and controls
> ... 
> I would like Terminal/PowerShell to behave exactly like Notepad

This seems like a reasonable property to add to the `copy` action. 

```ts
class CopyActionArgs {
  singleLine: bool = false;
  copyFormatting = false;
  dismissSelection: bool = false;
}
```

_Originally suggested by @gaberiel44 in https://github.com/microsoft/terminal/discussions/15349_

> **Note**
> ## Walkthrough 

* Add a `Boolean DismissSelection` to `CopyTextArgs` in `src\cascadia\TerminalSettingsModel\ActionArgs.idl`
* Add `DismissSelection` to `COPY_TEXT_ARGS` in `ActionArgs.h`
* The actual copy args are retrieved in `TerminalPage::_HandleCopyText` (in `AppActionHandlers.cpp`), and passed to `TerminalPage::_CopyText`, which calls down to `TermControl::CopySelectionToClipboard`. 
  * In `CopySelectionToClipboard`, just don't `ClearSelection`.
* This arg will also need localization, in `ActionArgs.cpp`, `CopyTextArgs::GenerateName`, and `src\cascadia\TerminalSettingsModel\Resources\en-US\Resources.resw`

### PR DESCRIPTION
**Pull request by gonzalo-garcian**
## Summary of the Pull Request
Adds a dismiss selection option to the "copy" action.

## PR Checklist
- [x] Closes #15371
- [x] Tests added/passed
- [x] Documentation updated
   - If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: MicrosoftDocs/terminal#686
- [x] Schema updated (if necessary)

### CODE REVIEW

**Comment by gonzalo-garcian** [comment]
@microsoft-github-policy-service agree

File: src/cascadia/TerminalSettingsModel/ActionArgs.h [hunk 1]
```diff
@@ -99,8 +99,9 @@ private:                                                                    \
 // false, if we don't really care if the parameter is required or not.
 
 ////////////////////////////////////////////////////////////////////////////////
-#define COPY_TEXT_ARGS(X)                           \
-    X(bool, SingleLine, "singleLine", false, false) \
+#define COPY_TEXT_ARGS(X)                                       \
+    X(bool, DismissSelection, "dismissSelection", false, false) \
```
Comment (lhecker) [inline src/cascadia/TerminalSettingsModel/ActionArgs.h:103]: The current behavior is "true". Should we keep the default as true?

Comment (gonzalo-garcian) [reply src/cascadia/TerminalSettingsModel/ActionArgs.h:103]: I set DismissSelection to false by default due to personal preference, but I agree with you, it should be set to true by default.

Comment (Delano5000) [reply src/cascadia/TerminalSettingsModel/ActionArgs.h:103]: Good start

**Review by lhecker — APPROVED** [review]

**Review by carlos-zamora — CHANGES_REQUESTED** [review]
Looks great! I even tested it out myself. Thanks! There's a few more things you need to do before I sign off on this though:
- [x] update [the schema](https://github.com/microsoft/terminal/blob/main/doc/cascadia/profiles.schema.json)
- [x] fix the localization stuff

**Comment by carlos-zamora** [comment]
Also, what's the repro for this?

> The first Ctrl + C throw an exeception.

File: src/cascadia/TerminalSettingsModel/ActionArgs.cpp [hunk 2]
```diff
@@ -187,6 +187,15 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
...
 
+        if (DismissSelection())
+        {
+            ss << RS_(L"DismissSelectionCommandKey").c_str();
+        }
+        else
+        {
+            ss << RS_(L"DismissSelectionFalseCommandKey").c_str();
+        }
+
```
[the remark this answers is not in the document]
Comment (gonzalo-garcian) [reply src/cascadia/TerminalSettingsModel/ActionArgs.cpp:198]: The truth is that I have programmed it a bit by intuition. Where do these strings appear/be used for? I haven't had much time to investigate. I guess I'll wait for others to comment on this.

**Comment by gonzalo-garcian** [comment]
> 

I think it's a Visual Studio thing because I've done some testing and it only happens if you start typing in the terminal when Visual Studio hasn't fully loaded the debug menus yet. When I have time I'll upload some gifs showing it.

**Comment by gonzalo-garcian** [comment]
> Also, what's the repro for this?
> 
> > The first Ctrl + C throw an exeception.

I think it's a Visual Studio thing because I've done some testing and it only happens if you start typing in the terminal when Visual Studio hasn't fully loaded the debug menus yet. When I have time I'll upload some gifs showing it.

File: src/cascadia/TerminalSettingsModel/ActionArgs.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (carlos-zamora) [reply src/cascadia/TerminalSettingsModel/ActionArgs.cpp:198]: No problem! haha `GenerateName()` spits out the localized string version of the action. These get displayed in the command palette and in the actions page of the settings UI.

File: src/cascadia/TerminalSettingsModel/Resources/en-US/Resources.resw [hunk 3]
```diff
@@ -707,4 +707,4 @@
     <value>Search the web for selected text</value>
     <comment>This will open a web browser to search for some user-selected text</comment>
   </data>
-</root>
+</root>
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsModel/Resources/en-US/Resources.resw:710]: you can undo changes to this file

**Review by carlos-zamora — APPROVED** [review]
Thanks! I'll go ahead and resolve the comment I made and clean up the PR body before merging it. Nice work 😊

**Review by DHowett — CHANGES_REQUESTED** [review]
taking some time to look at this myself

**Review by zadjii-msft — APPROVED** [review]
Exactly what I would have done, thanks!

**Comment by zadjii-msft** [comment]
image

**Pull request merged**